snapshot/process.go

```diff
@@ -64,35 +86,19 @@ func chooseProcessor(ctx context.Context, logger *logrus.Entry,
 			}
 		default:
 			// OCI image is also marked with "containerd.io/snapshot.ref" by Containerd
+			handler = defaultHandler
 		}
 	} else {
 		// Container writable layer comes into this branch. It can't be committed within this Prepare
 
-		remoteHandler := func(id string, labels map[string]string) func() (bool, []mount.Mount, error) {
-			return func() (bool, []mount.Mount, error) {
-				logger.Debugf("Found nydus meta layer id %s", id)
-				if err := sn.prepareRemoteSnapshot(id, labels); err != nil {
-					return false, nil, err
-				}
-
-				// Let Prepare operation show the rootfs content.
-				if err := sn.fs.WaitUntilReady(id); err != nil {
-					return false, nil, err
-				}
-
-				mounts, err := sn.remoteMounts(ctx, s, id)
-				return false, mounts, err
-			}
-		}
-
 		// Hope to find bootstrap layer and prepares to start nydusd
 		// TODO: Trying find nydus meta layer will slow down setting up rootfs to OCI images
 		if id, info, err := sn.findMetaLayer(ctx, key); err == nil {
 			logger.Infof("Prepares active snapshot %s, nydusd should start afterwards", key)
 			handler = remoteHandler(id, info.Labels)
 		}
 
-		if sn.fs.ReferrerDetectEnabled() {
+		if handler == nil && sn.fs.ReferrerDetectEnabled() {
```

Comment:
Is the `handler == nil` check right? It seems to have `handler = skipHandler` for the referrer.

Reply:
The `handler` variable is initialized to nil and it hasn't been assigned to any value for referrer case. So seems the check is right.